File: Dante1/Module1.08/Zadanie8.02/main.c
#include <stdio.h>

long power_rec(int a, int n);

int main()
{
    int a, n;

    printf("Enter a: ");

    int check1 = scanf("%i",&a);
    if(check1 != 1)
    {
        printf("Incorrect input");
        return 1;
    }

    printf("Enter n: ");

    int check2 = scanf("%i",&n);
    if(check2 != 1)
    {
        printf("Incorrect input");
        return 1;
    }
    if(n < 0)
    {
        printf("Incorrect input data");
        return 2;
    }
    if(n == 1)
    {
        printf("%i",a);
        return 0;
    }
    printf("%ld", power_rec(a, n));

    return 0;
}

long power_rec(int a, int n)
{
    if(n < 0)
    {
        return 0;
    }
    if(n < 1)
    {
        return 1;
    }
    return a * power_rec(a, n - 1);
}

File: Dante1/Module1.08/Zadanie8.02/main.rs
fn power_rec(a: u64, b: i32) -> u64
{
    if b < 0
    {
        return 0;
    }
    else if b < 1
    {
        return 1;
    }
    a * power_rec(a, b - 1)
}

fn main()
{
    let mut input: String = String::new();

    println!("Enter: ");

    std::io::stdin().read_line(&mut input).expect("Incorrect input");
    let a: u64 = input.trim().parse().expect("Incorrect parse");

    input.clear();

    println!("Enter: ");

    std::io::stdin().read_line(&mut input).expect("Incorrect input");
    let b: i32 = input.trim().parse().expect("Incorrect parse");
    if b < 0
    {
        panic!("Incorrect input data");
    }
    if b == 1
    {
        println!("Result is: {}", a);
        return;
    }

    println!("Result is: {}", power_rec(a, b));
}
